[20417] [Accessibility] The notification which tab of the split view a user is located on is unclear

HDinger opened the pull request:
This PR fixes a bug which occurs on the hidden label documenting the location of the use in WP split screen. Before the type was undefined. Further the same label was added for the right panel in details view.

https://community.openproject.com/work_packages/20417/activity

ulferts committed:
allow every char in the routes for a wiki

ulferts committed:
remove no longer necessary constraint on the wiki page title

ulferts committed:
consolitate wiki paths

ulferts committed:
remove no longer accessible wikis mgmt functionality

oliverguenther committed:
Collapse main menu early on loading the DOM

This really is a hack around the initialization of angular for the main
menu. Since the state of the menu is determined through sessionStorage,
we can leverage the immediate DOM placement to modify the menu before
the full page has been loaded.

Otherwise, the main menu flickers when collapsed (i.e., is first loaded
expanded and when the page is loaded, is being collapsed).

ulferts committed:
fix default query initialization again

oliverguenther committed:
Fix all karma tests exepect group sums

oliverguenther committed:
Use rubocop 2.1+

ulferts committed:
adapt sum specs

* as we are no longer needing total-sums.directive it is deleted

oliverguenther committed:
Add separate experimental and V3 mocks

This should fix protractor tests in the process.

ulferts committed:
correctly handle inline edit for attributes of type "User"

ulferts committed:
prevent trying to get into edit mode when user is not allowed to edit

ulferts committed:
remove unnecessary code

There will never be a params[:format] on this controller so we don't need a case statement for it

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Watch for attribute changes in the resource

oliverguenther committed:
Merge branch 'release/5.0' into dev

ulferts committed:
use camelCase for queryId

oliverguenther committed:
Show default placeholder when no value in td

oliverguenther committed:
Track objects by href and set self link

This adds the self href attribute to a HALResource whenever the object
has a self link.

This is used for tracking, e.g., selected objects.

ulferts committed:
limit which fields are editable

ulferts committed:
rename parameter in spec

ulferts committed:
adding version and category to lists of supported inline edit types

oliverguenther committed:
Track by 'null' object which is transformed before saving

The track by causes a 'null' href to land in the wp resource for changed links
which is not accepted by the API.

ulferts committed:
ensure the schema to be loaded witin a row

ulferts committed:
Merge pull request #4159 from furinvader/experimental/new_wp_table/1

WP list edit feature

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Fix webpack typing warnings

oliverguenther committed:
Fix _this in forEach

Use an arrow function instead of a regular function.

ulferts committed:
split existing edit field implementations into separate files

ulferts committed:
split up wp-edit-field.directive.html into separate files

ulferts committed:
Merge pull request #4213 from oliverguenther/fix/typescript-typing-warnings

Fix webpack typing warnings

ulferts committed:
add bool, float and integer cf to wp-edit

oliverguenther committed:
Merge pull request #4216 from ulferts/fix/separate_wp-edit-field_files

Fix/separate wp edit field files

ulferts committed:
extend wp-edit to work for StringObjects

oliverguenther committed:
Merge pull request #4214 from oliverguenther/fix/this-in-foreach

Fix _this in forEach

oliverguenther committed:
Allow plugins to receive field classes

Costs cannot directly import, e.g., the SelectField, so we want the
service to be able to return the registered classes in order for Costs
to register the SelectField for a new type (costObject).

oliverguenther committed:
Disable workaround in test mode

oliverguenther committed:
Bump byebug and pry

This bumps pry/byebug et al. due to issues with older versions on ruby
2.3

- pry
- pry-rails
- pry-stack_explorer
- pry-rescue
- pry-byebug
- pry-doc byebug

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge pull request #4219 from oliverguenther/bump/byebug

Bump byebug and pry

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Fix tests

oliverguenther committed:
Ignore default Firefox landing page

oliverguenther committed:
Remove 4.1 workaround

ulferts committed:
inplace edit fields of type 'Date'

This includes start/due date as well as custom fields.

oliverguenther committed:
Add HalTransform for API errors

oliverguenther committed:
Add HalTransform for API errors

oliverguenther committed:
Add column for attributes when in error

oliverguenther committed:
Add column for attributes when in error

furinvader committed:
Remove redundant source property from HalResource

oliverguenther committed:
Add very basic error styling

oliverguenther committed:
Add very basic error styling

oliverguenther committed:
Add separate ErrorResource

oliverguenther committed:
Add separate ErrorResource

furinvader committed:
Add apiV3 service tests

oliverguenther committed:
Properly handle no-longer erroneous fields

oliverguenther committed:
Properly handle no-longer erroneous fields

oliverguenther committed:
Add class states to outer td element

Mark the td element with `-editable` when the wp-edit field is editable.

oliverguenther committed:
Add class states to outer td element

Mark the td element with `-editable` when the wp-edit field is editable.

furinvader committed:
Remove Restangular methods from HalResource $source

furinvader committed:
Add tests for the _plain property of resources

oliverguenther committed:
Reset error state when submission successful

oliverguenther committed:
Reset error state when submission successful

HDinger committed:
Use styling of inplaceEditing for table edit

HDinger committed:
Use styling of inplaceEditing for table edit

oliverguenther committed:
Write feature spec

oliverguenther committed:
Let cursor appear over whole line

furinvader committed:
Make $source public

furinvader committed:
Refactor the HalResource a little

HDinger committed:
Fix bug with column width and avoid overflow of text

HDinger committed:
Add line seperator for dropdown menus in sidebar

furinvader committed:
Make properties of $links and $embedded non-settable

ulferts committed:
focus inplace edit fields

ulferts committed:
remove wp-edit directive instance if field is not active

furinvader committed:
Return empty object if the resource's source does not provide _links or _embedded

ulferts committed:
display placeholder on empty columns

This becomes important when displaying work packages with different types where one type does not have a custom field but that custom field is selected to be displayed within the table

ulferts committed:
Merge pull request #4195 from ulferts/feature/dots_in_wiki_page_titles

Feature/dots in wiki page titles

ulferts committed:
Merge pull request #4208 from oliverguenther/housekeeping/production-log-level

Use :warn as production default log level

furinvader committed:
Proxy the properties of the object to the source

furinvader committed:
Refactor $links

furinvader committed:
Refactor $embedded

furinvader committed:
Implement test for the href property

oliverguenther committed:
Bump oj

openprojectci committed:
Update reference to OpenProject-Translations

ulferts committed:
Merge pull request #4229 from oliverguenther/bump/gemfile

Bump dependencies

oliverguenther committed:
Merge pull request #4228 from ulferts/feature/add_wp-edit_custom_fields

Feature/add wp edit custom fields

oliverguenther committed:
Add duration field

HDinger committed:
Adapt datepicker styling in wp table

ulferts committed:
fix class selector determining when to hide the date picker

Without it, the date picker would be closed when the user tries to select a date

oliverguenther committed:
Add integration test

oliverguenther committed:
Merge pull request #4232 from ulferts/fix/inline_edit_date_picker

fix class selector determining when to hide the date picker

oliverguenther committed:
Merge pull request #4231 from HDinger/feature/22554-inline-edit-date-picker

[22554] Styling inplace edit date picker

ulferts committed:
Merge pull request #4218 from oliverguenther/feature/cost-object

[22559] Allow plugins to receive field classes

ulferts committed:
make required parameter explicit

furinvader committed:
Make property links lazy

ulferts committed:
initial request spec for editing project on wp resource

ulferts committed:
add project to payload representer

ulferts committed:
add specs for other linked attributes on wp payload resource

HDinger committed:
Introduce -small class for smaller inplace editing fields

ulferts committed:
Merge pull request #4230 from oliverguenther/feature/inline-edit-duration

Add duration field

furinvader committed:
Make the links lazy

ulferts committed:
define project changing behavior in wp contract

oliverguenther committed:
Separate wp-list controller query loading into services

oliverguenther committed:
Fix integration test after correct blur'ing

furinvader committed:
Make the embedded properties lazy

oliverguenther committed:
Remove double TS import

ulferts committed:
use variable in shared context

furinvader committed:
Make proxied properties enumerable

HDinger committed:
Insert separator even without hover or selection

furinvader committed:
Make the properties of the embedded elements lazy

furinvader committed:
Refactor embedded array element transformaiton

HDinger committed:
Introduce new color variable for the toggle separator

ulferts committed:
allow setting new custom field values when changing the project

oliverguenther committed:
Merge pull request #4227 from HDinger/feature/22845-add-line-to-separate-work-package-and-wiki-button

[22845] Add line to separate work package and wiki button

furinvader committed:
Refactor HalResource

furinvader committed:
Add make 'lazy' function a service

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Remove PaginationService setting from params

ulferts committed:
add block to controll when a notifiation is send

ulferts committed:
add capability to move work package accross projects to service

oliverguenther committed:
Load schema to determine writable state

oliverguenther committed:
Skip activation when already activated

ng-aria probably adds keypress handlers to ng-click handlers for
screenreaders, such as for space. This triggers `activate` of the inline
edit and causes the pristine copy to be invalid.

ulferts committed:
add service specific for adding notes to a work package

furinvader committed:
Rename hal sub folder to hal-resources

oliverguenther committed:
Very crudely reload table

oliverguenther committed:
Merge pull request #4225 from oliverguenther/feature/expand-error-columns

[22556] Error handling for other required/invalid fields of a WP row

oliverguenther committed:
Make escape cancel edition and restore pristine value

oliverguenther committed:
Copy live work packages based on lockVersion

oliverguenther committed:
Merge pull request #4233 from oliverguenther/feature/escape-cancels

Make escape cancel edition and restore pristine value

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Use restangular#withConfig to initialize uncached provider

oliverguenther committed:
Fix schema loading

oliverguenther committed:
Wrap $http for caching

We'll need an option to disable caching for specific routes, or ideally,
specify other means of caching requests.

This commit adds:
  - A wrapper for $http to catch caching options passed through
  Restangular
  - An improved CacheService that can handle loading promises
  (previously, it interfaced only to Resources)

HDinger committed:
Merge remote-tracking branch 'origin/release/5.0' into dev

HDinger committed:
Set tabindex to avoid mutliple tabs for one field

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge pull request #4240 from HDinger/fix/19689-details-pane-relations-tab-focus-is-not-set-on-input-field-when-expanding-field

[19689] In details pane in relations tab, the focus is not set correctly on input field when expanding field

ulferts committed:
click on the only space left for selecting a row

ulferts committed:
be more generous when waiting for the loading indicator

oliverguenther committed:
Very basic animation for removed rows

ulferts committed:
reapply b52b88a

oliverguenther committed:
Disable cache for queries

ulferts committed:
remove unused method

oliverguenther committed:
Merge pull request #4241 from ulferts/fix/select_work_package_row_spec

Fix/select work package row spec

oliverguenther committed:
Cache resources in halResource

manuschiller committed:
[22752] show all fields in edit mode

ulferts committed:
add resource for querying the list of allowed projects

oliverguenther committed:
Remove uncachedProvider

We'll no longer need that one

oliverguenther committed:
Keep loading Promise in Hal resource

oliverguenther committed:
Add Internet Explorer 10 as unsupported

ulferts committed:
update documentation to reflect changes to wp's project

manuschiller committed:
[22624] Add link to success message on work package update/create

manuschiller committed:
[22624] Add link to success message on work package update/create

ulferts committed:
add Project as an editable field in the edit field config

ulferts committed:
move contracts from lib/api to app/contracts

oliverguenther committed:
Work package table: Add link to id column

ulferts committed:
replace under the hood expectations in feature spec by observable behaviour

oliverguenther committed:
Merge pull request #4247 from oliverguenther/feature/table-id-link

Work package table: Add link to id column

oliverguenther committed:
Remove feature-flags

We no longer disable detailsView, thus save the extra request.

oliverguenther committed:
Merge pull request #4246 from ulferts/feature/edit_work_packages_project

Feature/edit work packages project

oliverguenther committed:
Remove automatic opening of split pane

furinvader committed:
Replace _$loadingPromise with $self and make it public

furinvader committed:
Remove TODO comments

oliverguenther committed:
Add split view button to hovered rows

ulferts committed:
extract contract related methods of service into concern

ulferts committed:
remove offending is_*?

HDinger committed:
Change styling of split view icon in wp table

ulferts committed:
add scope to Type to query for those enabled in a project

HDinger committed:
Change cursor and shrink line height of wp inline edit fields

oliverguenther committed:
Merge pull request #4245 from oliverguenther/housekeeping/ie10-unsupported-browser

Add Internet Explorer 10 as unsupported

ulferts committed:
fix contract path in op_engine

ulferts committed:
improve CreateWorkPackageService to conform with new interface

ulferts committed:
Remove project#add_work_package

openprojectci committed:
Update reference to OpenProject-Translations

furinvader committed:
Move error resource service to hal-resource folder

furinvader committed:
Fix work package save spec

furinvader committed:
Merge pull request #4238 from furinvader/feature/hal-resource-performance

Tweak HalResource Performance

oliverguenther committed:
Merge pull request #4257 from ulferts/fix/contract_path_in_op_engine

fix contract path in op_engine

oliverguenther committed:
Merge pull request #4253 from oliverguenther/housekeeping/remove-feature-flags

Remove feature-flags

HDinger committed:
Set cursor for ID field

ulferts committed:
add a project independent endpoint for work package creation

ulferts committed:
move CreateFormResource to CreateProjectFormResource

oliverguenther committed:
Merge pull request #4254 from oliverguenther/feature/wp-table/split-view-button

Add split view button on hovered work package rows

oliverguenther committed:
Merge pull request #4255 from HDinger/feature/inline-edit-cursor-line-hight

[18404] Styling inline edit: Change cursor and line height

oliverguenther committed:
Add fullscreen link to table context menu

ulferts committed:
explicitly require to avoid "still in tree" error

ulferts committed:
Handle empty create requests

furinvader committed:
Linked resources are now settable using HalResources as values

oliverguenther committed:
Merge pull request #4239 from oliverguenther/feature/inline-edit/reload-table

[22563] Reload and Merge WP table in background

oliverguenther committed:
Merge pull request #4258 from furinvader/fix/linked-properties

Linked resources are now settable using HalResources as values

oliverguenther committed:
Fix duration parsing in inline-edit

ulferts committed:
add a project independent endpoint to create work packages

Includes the auxillary end points:
* form
* schema
* available_projects

oliverguenther committed:
Inline edit: Add tooltip when form not editable

This adds a toolbar to attributes which are writable from their schemas,
but are no longer writable when their form is loaded.

This may happen for, e.g., priority of parent work packages, but also
other magic constraints.

While we work on reducing the magic in the constraints, a generic tooltip
offers the fastest user-friendly way of informing the user of
constraints.

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge pull request #4263 from oliverguenther/feature/inline-edit/22898-tooltip

[22898] Inline edit: Add tooltip when form not editable

oliverguenther committed:
Merge pull request #4261 from oliverguenther/fix/22829-duration-directive

[22829] Fix duration parsing in inline-edit

oliverguenther committed:
Remove private annotation for descendents of FormRepresenter

oliverguenther committed:
Set project and default type for project-scoped form

This was the default for new forms before and no type was set (e.g., creating a child)
and should not be broken.

HDinger committed:
Add additional condtition to avoid error message when clicking on ID

oliverguenther committed:
Specify new endpoints

Specifies the endpoints to be created by https://github.com/opf/openproject/pull/4262

[ci skip]

furinvader committed:
Remove parse error when opening details view

A little refactoring also took place to satisfy
the IDE - adding type definitions.
The distinction between arrow and ordinary
functions clarifies, where the arguments
object is needed and where it is not.

The useCaching variable always returned a
truthy value, because of the `|| true` at the
end. After removing it, the application startet
to function normally.

oliverguenther committed:
Merge pull request #4267 from oliverguenther/feature/independent-wp-doc

Specify new endpoints

oliverguenther committed:
Merge pull request #4262 from ulferts/feature/project_independent_work_package_create

Feature/project independent work package create

openprojectci committed:
Update reference to OpenProject-Translations

HDinger committed:
Merge pull request #4266 from HDinger/fix/avoid-error-message-when-clicking-on-id

[Inline Editing] Avoid error message when clicking on ID

oliverguenther committed:
Merge pull request #4268 from furinvader/fix/parse-error

Remove parse error when opening details view

oliverguenther committed:
Merge pull request #4249 from manuschiller/22752

[22752] show all fields in edit mode

HDinger committed:
Merge remote-tracking branch 'origin/release/5.0' into dev

oliverguenther committed:
Merge pull request #4252 from oliverguenther/feature/full-screen-context-item

Add fullscreen link to table context menu

HDinger committed:
Fix bug on wiki page

HDinger committed:
Remove underline from icon on hover

oliverguenther committed:
Merge pull request #4270 from HDinger/fix/inline-edit-icon-no-hover

[Inline Edit] Remove underline form split view icon in WP table

RobinWagner committed:
Change button text wiki

Change button text for wiki create button.

manuschiller committed:
Correct controlling when the link should be displayed and I18-Support

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge pull request #4251 from manuschiller/22624

[22624] Add link to success message on work package update/create

oliverguenther committed:
Fix Wiki const lookup

oliverguenther committed:
Merge pull request #4274 from oliverguenther/change-wiki-button-text

Change button text wiki

oliverguenther committed:
Close notifications before creating new ones

oliverguenther committed:
Merge pull request #4275 from oliverguenther/fix/specs

Fix current broken specs

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge branch 'release/5.0' into dev

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

furinvader committed:
Fix karma tests

* Exit karma with the given exit code after tests have run

* Fix NotificationBox directive test

* Set karma singleRun to true

* Refactor loadingIndicator service

* Fix apiV3 service tests

* Fix HalLink service tests

* Fix HalResource tests

* Reformat WP list controller test code

* Fix WP list controller tests

* Refactor wpContextMenu and fix its tests

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
[22916] Inline edit only on attribute text, double click to fullscreen

* Reduce editable field to wp-table--cell-span

* Allow macs to CMD-click for multiselect

* Set double click to full screen

* Set focus expectation to anchor

* Fix flickering in inline expectation

* Update selector for double clicks

* Simplify innerMostSpan

* Add spec for double click full view

manuschiller committed:
[20140] modals: fixes for accessibility of modals by keyboard

oliverguenther committed:
Avoid submitting an invalid query with empty status

oliverguenther committed:
Merge layouts into one

Use angular.html.erb as a nested layout which renders base itself.

oliverguenther committed:
Merge pull request #4211 from oliverguenther/fix/22704-navigation-flicker

[22704] Collapse main menu early on loading the DOM

oliverguenther committed:
[13702]  Default columns: Move subject left, remove priority

This moves the subject as a default column directly next to Type,
and removes prirority from the default set of columns.

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Merge branch 'release/5.0' into dev

openprojectci committed:
Update reference to OpenProject-Translations

HDinger committed:
Change field widths (#4312)

HDinger committed:
Set correct type and hidden label

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://hdinger-fix-20417-the-notification-which-tab-of-the-split-view-.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://hdinger-fix-20417-the-notification-which-tab-of-the-split-view-.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

oliverguenther committed:
Pass the decorated work package to CreateService (#4317)

Since associations will not be marked `dirty` when they're written by
the decorator, we are missing custom field values when creating new work
packages.

This reverts some of the changes introduced by the create service and
passes the decorated work package directly.

oliverguenther committed:
Merge branch 'release/5.0' into dev

HDinger committed:
Set html titles for wiki and wp pages (#4311)

Closely related to #4308 this PR sets the html titles for wiki and WP pages.

oliverguenther committed:
Avoid RuntimeError when aggregating with nil representers (#4321)

When aggregating columns, e.g., a custom field that contains nil values,
the AggregationGroup receives an empty represented.

This commit fixes that by allowing the represented object to be nil

```
RuntimeError (no represented object passed):
  lib/api/decorators/single.rb:54:in `initialize'
  lib/api/decorators/aggregation_group.rb:39:in `initialize'
  lib/api/v3/work_packages/work_package_list_helpers.rb:138:in `new'
  lib/api/v3/work_packages/work_package_list_helpers.rb:138:in `block in generate_groups'
  lib/api/v3/work_packages/work_package_list_helpers.rb:132:in `each'
  lib/api/v3/work_packages/work_package_list_helpers.rb:132:in `map'
  lib/api/v3/work_packages/work_package_list_helpers.rb:132:in `generate_groups'
  lib/api/v3/work_packages/work_package_list_helpers.rb:127:in `apply_and_generate_groups'
  lib/api/v3/work_packages/work_package_list_helpers.rb:42:in `work_packages_by_params'
  lib/api/v3/work_packages/work_packages_by_project_api.rb:42:in `block (2 levels) in <class:WorkPackagesByProjectAPI>'
  app/middleware/params_parser_with_exclusion.rb:38:in `call'
```

openprojectci committed:
Update reference to OpenProject-Translations

openprojectci committed:
Update reference to OpenProject-Translations

furinvader commented:
Closed as it became messy (sorry) - in favour of #4330
